network/server/tchannelthrift/node/service.go

```diff
@@ -207,6 +212,10 @@ func (s *service) FetchBlocksMetadata(tctx thrift.Context, req *rpc.FetchBlocksM
 			blockMetadata := rpc.NewBlockMetadata()
 			blockMetadata.Start = xtime.ToNanoseconds(fetchedMetadataBlock.Start())
 			blockMetadata.Size = fetchedMetadataBlock.Size()
+			if checksum := fetchedMetadataBlock.Checksum(); checksum != nil {
```

Comment:
There can be a variety of reasons, e.g., blocks not sealed, or the `includeChecksum` flag is set to false in the RPC call (which we don't do now but might do if the sizes are good enough and we want to save some bandwidth and processing power), or some error happened when reading/decoding the block's data.